src/cr/cube/cubepart.py

```diff
@@ -122,6 +122,11 @@ def population_fraction(self):
         """population fraction of the cube"""
         return self._cube.population_fraction
 
+    @lazyproperty
+    def selected_categories_label(self):
+        """Set of selected categories for all the dimensions."""
+        return {}
```

Comment:
I think this property shouldn't be here. I'm not exactly sure which class it's in (looking at diff only), but my guess is that that class shouldn't have that property. Only the class that can actually return something useful should have it (and then the legit return value would be `None`). And then we'd inspect it from the exporter, if it's exactly that combination of dimensions (or whatever the condition is) and _know_ that we need to print it out.

Reply:
Also, `{}` is an empty `dict`, not a set. An empty set is `set()`, although a `lazyproperty` should only ever be returning an (immutable) `frosenset()`.

Reply:
Also, I'm thinking `selected_category_labels` is grammatically more correct. This is a collection and so should be plural, not a single "label".

Reply:
A method with no implementation (by design, in a base class) should raise `NotImplementedError` like this:

```python
raise NotImplementedError(
    "`%s` must implement `.selected_category_labels`
    % self.__class__.__name__
)
```

This way, if this method is called on `_Slice` for example, and `_Slice` has not implemented it, you get the error message: "`_Slice` must implement `.selected_category_labels`", which is very clear about what needs to be done.

This allows you to mock the method on the base class, while protecting against that method implementation actually ever being inherited directly by a subclass.

Reply:
After chatting with Mike, we're thinking `.selected_category_labels` is a bit too general and could mean a lot of things. Better would be `.selected_ca_category_labels`, which this should be unless we can think of something better, but also the docstring should be robust, with minimally a paragraph that explains that this only applies in the "transforming a CA into MR" case and explaining how that works and when it arises in a business context.

Reply:
@ernestoarbitrio what about elaborating the docstring to educate the unfamiliar reader (all of us after about two weeks) about what selected CA categories are and the gist of the whole "make a CA into an MR" bit?